src/sfdt/__tests__/replaceSfdtBookmarks.test.js

```diff
@@ -0,0 +1,99 @@
+import get from 'lodash/get';
+import {getSFDT, getInline, getInlines} from '../../__tests__/utils';
+import tableInlines from './fixtures/tableInlines';
+
+import replaceSfdtBookmarks from '../replaceSfdtBookmarks';
+
+const data = {
+	K1: '123'
+};
+
+const data1 = {
+	K2: 'def'
+};
+
+const inlines = getInlines();
+const sfdt = getSFDT(inlines);
+
+// test for field data
+describe('sfdt bookmark modify', function() {
+	test('replaceSfdtBookmarks', function() {
+		// calculate length of inlines array
+		const originalLength = getInline(sfdt).length;
+
+		// run function we are testing
+		const result = replaceSfdtBookmarks(data, sfdt);
+		const currentInlines = getInline(result);
+
+		// make sure we get the same length of inlines, no addition, no subtraction
+		expect(currentInlines.length).toEqual(originalLength);
+		// check replacement went well
+		expect(currentInlines[1].name).toEqual('DATA::123');
+		expect(currentInlines[3].name).toEqual('DATA::123');
+	});
+
+	test('when data already exists', function() {
+		const sfdtWithData = {
+			sections: [
+				{
+					blocks: [
+						{
+							inlines: [...inlines]
+						}
+					]
+				}
+			]
+		};
+		const originalLength = getInline(sfdtWithData).length;
+		sfdtWithData.sections[0].blocks[0].inlines[1].name = 'DATA::123';
+
+		const result = replaceSfdtBookmarks(data, sfdtWithData);
+		// get results we want to check
+		const currentInlines = getInline(result);
+
+		// make sure we get the same length of inlines, no addition, no subtraction
+		expect(currentInlines.length).toEqual(originalLength);
+		// check replacement went well
+		expect(currentInlines[1].name).toEqual('DATA::123');
+	});
+
+	test('call function and replace bookmarks', function() {
```

Comment:
you are missing some tests on not happy path use cases, like you have 2 fields on original documents where 1 have its id including the other one id that you replace

ie sfdt has `abc` and `abcd`
and you replace `abc` by `def`
`abcd` should not become `abcdefd`